connectors/http/http-base/src/main/java/io/camunda/connector/http/base/request/apache/bulider/parts/ApacheRequestBodyBuilder.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.http.base.request.apache.bulider.parts;
+
+import static org.apache.hc.core5.http.ContentType.APPLICATION_FORM_URLENCODED;
+import static org.apache.hc.core5.http.HttpHeaders.CONTENT_TYPE;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.camunda.connector.api.json.ConnectorsObjectMapperSupplier;
+import io.camunda.connector.http.base.model.HttpCommonRequest;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.text.StringEscapeUtils;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.io.entity.HttpEntities;
+import org.apache.hc.core5.http.io.entity.StringEntity;
+import org.apache.hc.core5.http.io.support.ClassicRequestBuilder;
+import org.apache.hc.core5.http.message.BasicNameValuePair;
+
+/**
+ * Maps the request body of a {@link HttpCommonRequest} to an Apache {@link ClassicRequestBuilder}.
+ */
+public class ApacheRequestBodyBuilder implements ApacheRequestPartBuilder {
+
+  @Override
+  public void build(ClassicRequestBuilder builder, HttpCommonRequest request)
+      throws JsonProcessingException {
+    if (request.getMethod().supportsBody && request.hasBody()) {
+      unescapeBody(request);
+
+      switch (request.getBody()) {
+        case Map<?, ?> body -> {
+          if (isFormUrlEncoded(request)) {
+            setUrlEncodedFormEntity(body, builder);
+          } else {
+            setStringEntity(builder, request);
+          }
+        }
+        case List<?> ignored -> setStringEntity(builder, request);
+        case String ignored -> setStringEntity(builder, request);
+        default ->
+            throw new IllegalStateException(
+                "Unexpected value type for the request body. Excepting Map, List, or String, got "
+                    + request.getBody().getClass().getSimpleName());
```

Comment:
To my knowledge, FEEL engine can return [other types](https://docs.camunda.io/docs/components/modeler/feel/language-guide/feel-data-types) including primitives (numbers, booleans) and dates. Although obviously not a super common use case, I can imagine that someone puts a FEEL expression into the body that evaluates to one of these types.

Reply:
Into the body I agree, but do you think the whole `body` property could be one the types you mentioned?

Could you provide an example if that's the case?

Reply:
I checked date, boolean, and a number - luckily, dates already arrive to us as strings at this point, but things like these two below result in numbers/booleans accordingly

<img width="387" alt="Screenshot 2024-05-31 at 19 00 10" src="https://github.com/camunda/connectors/assets/38818382/c70f6447-3dde-49a8-bb11-f2483922b5c2">
<img width="380" alt="Screenshot 2024-05-31 at 18 59 50" src="https://github.com/camunda/connectors/assets/38818382/df671d71-0cf9-449d-b556-9ac2d933f545">